core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -90,37 +90,46 @@ object ConsumerGroupCommand {
 
     protected def opts: ConsumerGroupCommandOptions
 
-    protected def getLogEndOffset(topic: String, partition: Int): LogEndOffsetResult
+    protected def getLogEndOffset(topic: Option[String], partition: Option[Int]): LogEndOffsetResult
 
     protected def describeGroup(group: String): Unit
 
-    protected def describeTopicPartition(group: String,
+    protected def describeMemberTopicPartition(group: String,
+                                         topic: String,
                                          topicPartitions: Seq[TopicAndPartition],
                                          getPartitionOffset: TopicAndPartition => Option[Long],
-                                         getOwner: TopicAndPartition => Option[String]): Unit = {
-      topicPartitions
-        .sortBy { case topicPartition => topicPartition.partition }
-        .foreach { topicPartition =>
-          describePartition(group, topicPartition.topic, topicPartition.partition, getPartitionOffset(topicPartition),
-            getOwner(topicPartition))
-        }
+                                         getMemberId: TopicAndPartition => Option[String],
+                                         getOwner: TopicAndPartition => Option[String],
+                                         getOwnerHost: TopicAndPartition => Option[String]): Unit = {
```

Comment:
Could we change getOwner and getOwnerHost to getClientId and getMemberHost accordingly? Also, I am not sure why we getMemberId, getOwner and getOwnerHost need to be a function. Could we just pass in an Option and get rid of "get"?